config/config.go

```diff
@@ -8,56 +8,99 @@ import (
 	"errors"
 	"flag"
 	"fmt"
-	"io/ioutil"
+	"os"
 
 	"github.com/elastic/elastic-agent-client/v7/pkg/client"
 	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
 	"github.com/elastic/elastic-agent-libs/config"
 	"github.com/elastic/elastic-agent-libs/logp"
-	"github.com/elastic/go-ucfg/json"
 
 	"github.com/elastic/elastic-agent-shipper/monitoring"
 	"github.com/elastic/elastic-agent-shipper/output"
 	"github.com/elastic/elastic-agent-shipper/queue"
-	"github.com/elastic/elastic-agent-shipper/server"
+
+	"go.uber.org/zap/zapcore"
 )
 
 var (
-	configFilePath    string
+	configFilePath string
+	// ErrConfigIsNotSet reports that no unmanaged config file has been set
 	ErrConfigIsNotSet = errors.New("config file is not set")
+	fs                = flag.CommandLine
+	// Overwrites is the config map of CLI overwrites set by the -E flag
+	Overwrites = config.SettingFlag(fs, "E", "config overwrites")
+
+	esKey      = "elasticsearch"
+	consoleKey = "console"
+	kafaKey    = "kafka"
 )
 
 // A lot of the code here is the same as what's in elastic-agent, but it lives in an internal/ library
 func init() {
-	fs := flag.CommandLine
 	fs.StringVar(&configFilePath, "c", "", "Run the shipper in the unmanaged mode and use the given configuration file instead")
+
+}
+
+// ShipperClientConfig is the shipper-relevant portion of the config received from input units
+type ShipperClientConfig struct {
+	Server string     `config:"server"`
+	TLS    ShipperTLS `config:"ssl"`
+}
+
+// ShipperTLS is TLS-specific shipper client settings
+type ShipperTLS struct {
+	CAs  []string `config:"certificate_authorities"`
+	Cert string   `config:"certificate"`
+	Key  string   `config:"key"`
 }
 
-// ShipperConfig defines the options present in the config file
+// ShipperRootConfig defines the shipper config we get from elastic-agent's output unit
+type ShipperRootConfig struct {
+	Type    string        `config:"type"`
+	Shipper ShipperConfig `config:"shipper"`
+}
+
+// ShipperConfig defines the config values stored under the `shipper` key in the fleet config
 type ShipperConfig struct {
-	Log     logp.Config       `config:"logging"`
 	Monitor monitoring.Config `config:"monitoring"` //Queue monitoring settings
 	Queue   queue.Config      `config:"queue"`      //Queue settings
-	Server  server.Config     `config:"server"`     //gRPC Server settings
 	Output  output.Config     `config:"output"`     //Output settings
+	// StrictMode means that every incoming event will be validated against the
+	// list of required fields. This introduces some additional overhead but can
+	// be really handy for client developers on the debugging stage.
+	// Normally, it should be disabled during production use and enabled for testing.
+	// In production it is preferable to send events to the output if at all possible.
+	StrictMode bool                `config:"strict_mode"`
+	Server     ShipperClientConfig // server settings, set by the input unit
```

Comment:
Having the name "Server" and "ShipperClientConfig" being matched up seems awkward to me.  Maybe change "ShipperClientConfig" to something like "ShipperConnectionConfig"?